src/rpc/blockchain.cpp

```diff
@@ -518,6 +518,69 @@ UniValue getblockbynumber(const UniValue& params, bool fHelp)
     return blockToJSON(block, pblockindex, params.size() > 1 ? params[1].get_bool() : false);
 }
 
+UniValue getblocksbatch(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() < 1 || params.size() > 3)
+        throw runtime_error(
+                "getblocksbatch <starting number> <number of blocks> [bool:txinfo]\n"
+                "\n"
+                "<starting number> the block number at the start of the batch"
+                "\n"
+                "<number of blocks> the number of blocks to return in the batch, limited to 500"
+                "\n"
+                "[bool:txinfo] optional to print more detailed tx info\n"
+                "\n"
+                "Returns a JSON array with details of the requested blocks starting with\n"
+                "the given block-number.\n");
+
+    UniValue result(UniValue::VOBJ);
+    UniValue blocks(UniValue::VARR);
+
+    int nHeight = params[0].get_int();
+    if (nHeight < 0 || nHeight > nBestHeight) throw runtime_error("Starting block number out of range");
+
+    int batch_size = params[1].get_int();
+    if (batch_size < 1 || batch_size > 500)
+    {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Batch size must be between 1 and 500, inclusive.");
+    }
+
+    bool transaction_details = false;
+    if (params.size() > 2) transaction_details = params[2].get_bool();
+
+    LOCK(cs_main);
+
+    // Select the block index for the head of the chain.
+    CBlockIndex* pblockindex_head = mapBlockIndex[hashBestChain];
+    CBlockIndex* pblockindex = pblockindex_head;
+
+    // Rewind to the block corresponding to the specified height.
+    while (pblockindex->nHeight > nHeight)
+    {
+        pblockindex = pblockindex->pprev;
+    }
+
+    int i = 0;
+    while (i < batch_size)
+    {
+        CBlock block;
+        if (!block.ReadFromDisk(pblockindex, true)) throw runtime_error("Error reading block from specified batch.");
```

Comment:
```suggestion
        if (!block.ReadFromDisk(pblockindex, transaction_details)) throw runtime_error("Error reading block from specified batch.");
```

Reply:
Actually, this doesn't work. Segfaults.

Reply:
Reverted back in the latest commit. I will check on ReadFromDisk later.

Reply:
`blockToJSON` seems to be not written with transaction_details being false in mind.

- L108, `block.GetMint()` doesn't crash but shouldn't provide sensible data as block.vtx is not populated.
- L110, `block.vtx[0]` doesn't exist.
- L132, `block.GetClaim()` empty vtx; can't get the claim.
- L167, vchBlockSig is not populated.
- L169, `block.GetClaim()` ditto.
- L171, `block.vtx[0]` doesn't exist.

Reply:
Yep. That needs to be fixed, but it is too much surgery to go into Ingrid. This PR is a completely self-contained change that is a new function and introduces no regression risks. BlockToJSON is used in other places, so we will straighten this out in another PR for Janice.

Reply:
Although the minimum provided by just providing the header only really doens't align with the intended JSON output leaving out only the transaction details, so maybe we just leave it alone. I will look at this more carefully later.